compaction/compaction_manager.cc

```diff
@@ -1198,14 +1198,20 @@ future<> compaction_manager::await_tasks(std::vector<shared_ptr<compaction_task_
 }
 
 std::vector<shared_ptr<compaction_task_executor>>
-compaction_manager::do_stop_ongoing_compactions(sstring reason, std::function<bool(const compaction_group_view*)> filter, std::optional<compaction_type> type_opt) noexcept {
+compaction_manager::do_stop_ongoing_compactions(sstring reason, std::function<bool(const compaction_group_view*)> filter, std::optional<compaction_type> type_opt, double stop_threshold) noexcept {
     auto ongoing_compactions = get_compactions(filter).size();
     auto tasks = _tasks
             | std::views::filter([&filter, type_opt] (const auto& task) {
                 return filter(task.compacting_table()) && (!type_opt || task.compaction_type() == *type_opt);
             })
             | std::views::transform([] (auto& task) { return task.shared_from_this(); })
             | std::ranges::to<std::vector<shared_ptr<compaction_task_executor>>>();
+    auto tasks_to_stop = tasks
+            | std::views::filter([stop_threshold] (const auto& task) {
+                    auto& cdata = task->compaction_data();
+                    return !cdata.compaction_size || (task->progress_monitor().get_progress() / cdata.compaction_size) < stop_threshold;
```

Comment:
there's a pitfall with how we calculate progress percentage. progress_monitor should be able to provide it. we trust that get_progress() and cdata.compaction_size will have the same unit in the future, but a change can break it. when creating the task, we could feed it with the estimated total work. not progress isn't bytes written, but rather bytes read (which one get_progress() returns?), since output can be much smaller than input size.